### PR TITLE
Removed BitGo dependency to mitigate security vulnerabilities

### LINKED ISSUE #26: remove legacy bitcoin client code

### LINKED ISSUE #49: set up travis
this repo now has tests! it should run the tests and require passing tests for merge

### LINKED ISSUE #54: security checks are failing
`npm run test` shows a security error:

```
> nsp check

(+) 1 vulnerabilities found
┌───────────────┬─────────────────────────────────────────────────────────────┐
│               │ Large gzip Denial of Service                                │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ Name          │ superagent                                                  │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ CVSS          │ 3.7 (Low)                                                   │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ Installed     │ 3.5.2                                                       │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ Vulnerable    │ <3.7.0                                                      │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ Patched       │ >=3.7.0                                                     │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ Path          │ bat-client@2.2.1 > bitgo@4.20.0 > superagent@3.5.2          │
├───────────────┼─────────────────────────────────────────────────────────────┤
│ More Info     │ https://nodesecurity.io/advisories/479                      │
└───────────────┴─────────────────────────────────────────────────────────────┘
```

### PR DESCRIPTION
**Pull request by jasonrsadler**
resolves #54 
resolves #49 
resolves #26 

per @evq, BitGo dependency is no longer used from BC POC.

Also added 'clean' script for rebuilds

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+language: node_js
+node_js:
+  - "7"
+dist: trusty
+sudo: required
+group: edge
+script:
+  - npm run test-security && npm run test-index
```
Comment (diracdeltas) [inline .travis.yml]: any reason not to also run `npm run test-integration`?

Comment (jasonrsadler) [reply .travis.yml]: No. Other than it's not fixed yet. :)

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+language: node_js
+node_js:
+  - "7"
```
Comment (diracdeltas) [inline .travis.yml]: according to https://github.com/nodejs/Release#release-schedule, Node 7 has been EOL'ed. current stable is Node 9 and LTS is Node 8. maybe just put "node" here to specify the latest stable release? (according to https://docs.travis-ci.com/user/languages/javascript-with-nodejs/#Specifying-Node.js-versions)

File: .travis.yml (hunk 1 above)
Comment (NejcZdovc) [reply .travis.yml]: @evq fixed this in #56, so we can add it as well

File: package.json [hunk 3]
```diff
@@ -32,13 +33,13 @@
...
     "bat-balance": "^1.0.7",
     "bat-publisher": "^2.0.9",
-    "bitgo": "^4.15.0",
     "brave-crypto": "^0.1.0",
     "http-request-signature": "0.0.2",
     "joi": "^13.1.2",
     "json-stable-stringify": "1.0.1",
     "node-anonize2-relic-emscripten": "^0.3.3",
     "random-lib": "2.1.0",
+    "snazzy": "^7.1.1",
```
Comment (NejcZdovc) [inline package.json]: please move this dependency to `devDependencies`

File: index.js [hunk 4]
```diff
@@ -644,28 +641,6 @@ Client.prototype.transition = function (newPaymentId, callback) {
     if (err) return callback(err)
 
     if (!body.unsignedTx) return callback(new Error('expecting an unsignedTx'))
```
Comment (evq) [inline index.js]: @jasonrsadler you can remove this entire `transition` function - it is no longer used in `browser-laptop`

**Comment by evq** [comment]
thanks for taking this on @jasonrsadler , left one comment

otherwise LGTM assuming pass when Travis is fully hooked up :D

File: .travis.yml (hunk 2 above)
Comment (NejcZdovc) [reply .travis.yml]: +1 for `node`

File: index.js (hunk 4 above)
Comment (jasonrsadler) [reply index.js]: Done. There is also 'transitioned ()'. Does that go as well or stay?

File: package.json (hunk 3 above)
Comment (jasonrsadler) [reply package.json]: Done

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+dist: trusty
+sudo: required
+group: edge
+script:
+  - npm run test-security && npm run test-index
+branches:
+  only:
+    - master
+    - /\d+\.\d+\.x/
```
[the remark this answers is not in the document]
Comment (jasonrsadler) [reply .travis.yml]: Removed

File: .travis.yml (hunk 2 above)
Comment (jasonrsadler) [reply .travis.yml]: Done

File: .travis.yml (hunk 1 above)
Comment (jasonrsadler) [reply .travis.yml]: Awesome! Done.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+language: node_js
+node_js:
+<<<<<<< HEAD
```
Comment (NejcZdovc) [inline .travis.yml]: oops, some merge problems

**Comment by jasonrsadler** [comment]
@evq one thing I did notice was that security tests would have one failure on clean run. Subsequent runs would pass.

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bat-client",
-  "version": "2.2.2",
+  "version": "2.2.3",
```
Comment (mrose17) [inline package.json]: before you do the "final commit":, please take a look at the master branch and make sure that you are bumping the version by one from whatever it is...

Comment (jasonrsadler) [reply package.json]: Done

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bat-client",
-  "version": "2.2.3",
+  "version": "2.2.4",
```
Comment (diracdeltas) [inline package.json]: You should run `npm install` after bumping this version so that it auto-updates the version in package-lock.json

**Review by diracdeltas — APPROVED** [review]
🥇

**Pull request merged**